tests/cli/workflow/rnafusion/test_cli_rnafusion_report_deliver.py

```diff
@@ -70,23 +73,41 @@ def test_without_samples(
     assert "no samples" in caplog.text
 
 
-def test_dry_run(
+def test_successful(
     cli_runner: CliRunner,
     rnafusion_context: CGConfig,
-    caplog: LogCaptureFixture,
     rnafusion_case_id: str,
+    rnafusion_deliverables_file_path: Path,
+    deliverables_template_content: List[dict],
+    mock_analysis_finish,
+    caplog: LogCaptureFixture,
+    mocker,
 ):
-    """Test command with case_id and analysis_finish which should execute successfully."""
+    """Test that deliverable files is properly created on a valid and successful run."""
     caplog.set_level(logging.INFO)
-    # GIVEN case-id
-    case_id: str = rnafusion_case_id
 
-    # WHEN dry running with dry specified
-    result = cli_runner.invoke(report_deliver, [case_id, "--dry-run"], obj=rnafusion_context)
+    # GIVEN a successful run
+
+    # GIVEN a mocked deliverables template
+    mocker.patch.object(
+        RnafusionDeliverables,
+        "get_deliverables_template",
+        return_value=deliverables_template_content,
+    )
+
+    # WHEN dry running
+    result = cli_runner.invoke(report_deliver, [rnafusion_case_id], obj=rnafusion_context)
```

Comment:
Do we dry run? Seems like we are not, and that we are expecting changes to files.

Reply:
🦅 👁️ ! good catch. This is wrong